packages/mangal/build.sh

```diff
@@ -0,0 +1,24 @@
+TERMUX_PKG_HOMEPAGE=https://github.com/metafates/mangal
+TERMUX_PKG_DESCRIPTION="Cli manga downloader"
+TERMUX_PKG_LICENSE="MIT"
+TERMUX_PKG_MAINTAINER="@termux"
+TERMUX_PKG_VERSION="3.9.0"
+TERMUX_PKG_SRCURL=https://github.com/metafates/mangal/archive/refs/tags/v${TERMUX_PKG_VERSION}.tar.gz
+TERMUX_PKG_SHA256=c6d818bc0ecc969020e2dc176053cb9068ea0363b4c0ab018ce1d070d098e4eb
+TERMUX_PKG_BUILD_IN_SRC=true
+TERMUX_PKG_AUTO_UPDATE=true
+
+termux_step_pre_configure() {
+	termux_setup_golang
+
+	go mod init || :
+	go mod tidy 
+}
+
+termux_step_make() {
```

Comment:
```shell
REVISION=$(git ls-remote https://github.com/metafates/mangal refs/tags/v${TERMUX_PKG_VERSION}^{} | head -c 7)
go build -ldflags "-X 'github.com/metafates/mangal/constant.BuiltAt=$(date -u)' -X 'github.com/metafates/mangal/constant.BuiltBy=$(whoami)' -X 'github.com/metafates/mangal/constant.Revision=${REVISION}' -s -w" -o mangal
```

Reply:
Makefile won't do this?

Reply:
> Makefile won't do this?

It will, but I tried building it myself and looks like archive is downloaded without `.git` folder which makes it's impossible to use `git rev-parse`. Maybe I did something wrong, dk, let me check once more

Reply:
I guess I could improve the Makefile to handle such cases instead 🤔

Reply:
Could also switch to using git to get sources: `TERMUX_PKG_SRCURL=https://github.com/metafates/mangal.git`

Reply:
> Could also switch to using git to get sources: `TERMUX_PKG_SRCURL=https://github.com/metafates/mangal.git`

Yeah, that's much better. Didn't know it was possible

Reply:
Ok I am doing it